File: caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{   //GETTING KEY
    if(argc!=2)
    {
        printf("usage: ./caesar  k\n");
        return 1;
    }
    int key= atoi(argv[1]);

    //GETTING PLAINTEXT
    string plaintext;

    plaintext= get_string("plaintext: ");



    //PRESERVING CASE AND ENCIPHERING
    printf("ciphertext: ");
    for(int i=0, n=strlen(plaintext); i<n; i++)
    {


        if (isupper(plaintext[i]))
        {
            printf("%c",((plaintext[i]-'A'+key)%26+'A'));
        }
        else if(islower(plaintext[i]))
        {
            printf("%c",((plaintext[i]-'a'+key)%26+'a'));
        }



        else
        {
            printf("%c",plaintext[i]);
        }

    }
    printf("\n");

}

File: caesar/caesar.py
from cs50 import get_string
import sys

# prompt key
if len(sys.argv) != 2:
    print("Usage:python caesar.py k")
    sys.exit(1)

k = int(sys.argv[1])

# prompt plaintext
p = get_string('Plaintext: ')
cipher = ''
# encipher
# for every character in plaintext
for i in p:

    # check if Alpha
    if i.isalpha() == True:

        # check if upper
        if i.isupper() == True:
            # set alphabet to A
            alpha = 'A'
        else:
            # set alphabet to a
            alpha = 'a'

        # subtract ascii of Alphabet from ascii of char,apply formula ,add ascii Alphabet
        c = chr((((ord(i) - ord(alpha)) + k) % 26) + ord(alpha))
        i = c

    # final string+charcter
    cipher += i

# print ciphertext
print('ciphertext: ' + cipher)

